File: ruby/engine/lib/taka_tic_tac_toe/player_factory.rb
module TakaTicTacToe
  class PlayerFactory
    def self.create(input)
      case input[:type]
      when :computer
        Computer.new(input[:mark])
      when :human
        Human.new(input[:mark])
      else
        raise "unknown player type"
      end
    end
  end
end

File: objective-c/console/ttt_engine/Src/Players/PlayerFactory.m
#import "PlayerFactory.h"
#import "Player.h"
#import "HumanPlayer.h"
#import "ComputerPlayer.h"

@implementation PlayerFactory

+ (id<Player>) create:(NSString*)playerType withMark:(NSString*)mark {
    if ([playerType isEqualToString:@"h"]) {
        return [[HumanPlayer alloc] init:mark];
    } else {
        return [[ComputerPlayer alloc] init:mark];
    }
}

@end
